mtags/src/main/scala-3/scala/meta/internal/pc/completions/OverrideCompletions.scala

```diff
@@ -494,12 +496,24 @@ object OverrideCompletions:
           case _ => false
       case _ => false
     val start = defn.span.start
-    val offset =
+    val braceOffset =
       if hasSelfTypeAnnot then text.indexOf("=>", start) + 1
       else text.indexOf("{", start)
-    if offset > 0 && offset < defn.span.end then Some(offset)
+    if braceOffset > 0 && braceOffset < defn.span.end then Some(braceOffset)
+    else hasColon(text, defn)
+  end hasBracesOrColon
+
+  private def hasColon(text: String, defn: TargetDef)(using
+      Context
+  ): Option[Int] =
+    val defnText = text.substring(defn.span.start, defn.span.end)
+    // If defn has an end marker, defnText must contain `:` and end with name
+    if defnText.endsWith(defn.name.decoded) then
+      val colonOffset = defnText.lastIndexOf(":")
+      if colonOffset > 0 then Some(defn.span.start + colonOffset)
+      else None
```

Comment:
```suggestion
      defn match
        case td: TypeDef if text.charAt(td.rhs.span.end) == ':'  =>
          Some(td.rhs.span.end)
        case _ => None
```

`td.rhs.span.end` points at `:` if there are no declarations, so this should be safe. You could also do 

```
      defn match
        case td: TypeDef if text.charAt(td.rhs.span.end) == ':'  && emptyDeclarations=>
          Some(td.rhs.span.end)
        case _ => None
```
with `emptyDeclarations` being forwarded from up the stack.